oauth2_http/java/com/google/auth/oauth2/GoogleCredentials.java

```diff
@@ -166,6 +168,24 @@ public static GoogleCredentials fromStream(
             fileType, USER_FILE_TYPE, SERVICE_ACCOUNT_FILE_TYPE));
   }
 
+  /**
+   * Adds quota project ID to requestMetadata if present.
+   *
+   * @return a new map with quotaProjectId added if needed, or else returns the original
```

Comment:
This is weird: sometimes it's a copy and sometimes it isn't? Pick one or the other. If this is only used internally, it's probably OK to modify the map.

Reply:
upstream the map unmodifiable. I will just always make a new map in this case then.